index.js

```diff
@@ -100,9 +87,15 @@ module.exports = function enforceHTTPS(options) {
       return next();
     }
 
-    // Check if method should be Forbidden
-    if (!isAllowed(ctx.method, options)) {
-      ctx.response.status = 403;
+    // Check if method should be disallowed (and handle OPTIONS method)
+    if (!redirectStatus[ctx.method]) {
+      if (ctx.method === 'OPTIONS') {
+        ctx.response.status = 200;
```

Comment:
Per spec, CORS relies on a different header (see https://www.w3.org/TR/cors/#access-control-allow-methods-response-header ). I don't know whether implementations fall back to Allow when the actual CORS header(s) are not available but the spec says not to.

RFC 7231, sections 4.3, 6.5, 6.6, 7.4 seem to be on-topic here. 4.3 is unfortunately lacking in guidance, though. I was mostly going on arguments from various parties that RESTful applications may want to preflight a request to see whether it's permissible or meaningful before actually making it; for that, the 200 response with Allow header and no body (or a configurable body explaining what the methods do—but that would be the job of another middleware) is useful.

In general, unless the request is denied by deliberate administrative decision, 403 is misleading. 405 or 501 would be better. 405 indicates a problem with the client's request, i.e., "this is not supported and don't try again"; 501 indicates a server issue, "this is not supported but might be in the future." This applies just as much to OPTIONS as to GET or HEAD; a failed response to OPTIONS should probably include an Allow header that doesn't allow OPTIONS!